src/commands/periscope/periscope.ts

```diff
@@ -0,0 +1,163 @@
+                await runAKSPeriscope(cluster, clusterKubeConfig);
+            }
+        } else {
+            vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+        }
+    }
+}
+
+async function runAKSPeriscope(
+    cluster: AksClusterTreeItem,
+    clusterKubeConfig: string
+): Promise<void | undefined> {
+    const clusterName = cluster.name;
+
+    // Get Diagnostic settings for cluster and get associated storage account information.
+    const clusterDiagnosticSettings = await longRunning(`Identifying cluster diagnostic settings.`,
+        () => getClusterDiagnosticSettings(cluster)
+    );
+
+    if (!clusterDiagnosticSettings || !clusterDiagnosticSettings.value?.length) {
+        // If there is no storage account attached to diagnostic setting, don't move forward and at this point we will render webview with helpful content.
+        await createPeriscopeWebView(cluster.name, undefined, undefined, false);
+        return undefined;
+
+    }
+
+    const clusterStorageAccountId = await longRunning(`Identifying cluster associated storage account.`,
+        () => chooseStorageAccount(clusterDiagnosticSettings)
+    );
+
+    // Generate storage sas keys, manage aks persicope run.
+    if (!clusterStorageAccountId) return undefined;
+
+    const clusterStorageInfo = await longRunning(`Generating SAS for ${clusterName} cluster.`,
+        () => getStorageInfo(cluster, clusterStorageAccountId)
+    );
+
+    if (!clusterStorageInfo) return undefined;
+
+    const aksDeplymentFile = await longRunning(`Deploying AKS Periscope to ${clusterName}.`,
+        () => prepareAKSPeriscopeDeploymetFile(clusterStorageInfo)
+    );
+
+    if (!aksDeplymentFile) return undefined;
+
+    const runCommandResult = await longRunning(`Running AKS Periscope on ${clusterName}.`,
+        () => runAssociatedAKSPeriscopeCommand(aksDeplymentFile, clusterKubeConfig)
+    );
+
+    await createPeriscopeWebView(clusterName, runCommandResult, clusterStorageInfo);
+
+}
+
+async function runAssociatedAKSPeriscopeCommand(
+    aksPeriscopeFile: string,
+    clusterKubeConfig: string | undefined
+): Promise<k8s.KubectlV1.ShellResult | undefined> {
+    const kubectl = await k8s.extension.kubectl.v1;
+
+    if (kubectl.available) {
+        // Clean up running instance.
+        await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+            clusterKubeConfig, "YAML",
+            (f) => kubectl.api.invokeCommand(`delete ns aks-periscope --kubeconfig="${f}"`));
+
+        // Deploy the aks-periscope.
+        const runCommandResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+            clusterKubeConfig, "YAML",
+            (f) => kubectl.api.invokeCommand(`apply -f ${aksPeriscopeFile} --kubeconfig="${f}" && kubectl cluster-info --kubeconfig="${f}"`));
+
+        return runCommandResult;
+    } else {
+        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
```

Comment:
Nit: again, this is the sort of case where I would consider early exit - dispose of the unlikely error case, and then proceed with the main logic without having an unfinished `if` dangling over the reader.

```
const kubectl = ...;

if (!kubectl.available) {
  // message and return
}

// now do all the stuff that uses kubectl unconditionally and without the extra indent!
```